CODEWALK.md

```diff
@@ -346,6 +346,118 @@ essentially who is storing what data, for what fee and which sectors have been s
 The keystore contains the binary encoded peer key for interacting securely over the network. 
 This data lives in a file at `$HOME/.filecoin/keystore/self`.
 
+## libp2p Networking
+
+As mentioned above, Filecoin relies on [libp2p](https://libp2p.io/) for its
+networking needs. Here we'll explain some of the mechanics that drive the
+network layer, in particular:
+
+1. Node discovery
+2. NAT topology determination
+3. Autorelay services for solving connectivity issues
+
+### Node discovery
+
+We use the libp2p Kademlia DHT bootstrapping mechanism to find Filecoin peers
+to interface with. libp2p maintains a k-bucket routing table, where peers are
+organized according to the _distance metric proposed_ in the [Kademlia
+paper](http://www.scs.stanford.edu/~dm/home/papers/kpos.pdf).
+
+To populate our routing table, and to keep it healthy over time, we traverse
+the network periodically (and once on startup) by performing random walks. We
```

Comment:
Unrelated, but this is a problem. Randomly walking when your routing table favours nodes closer to your address means you're getting almost no value to the routing table from this behaviour. Additionally, randomly walking in the bootstrap methods provides no way for the caller to benefit from the nodes seen. The correct approach if the nodes are to be arbitrarily connected to to maintain a mesh, would be to use a FindPeer or related call, and operate on the nodes visited.

Reply:
I've read the bit further down, it seems the random walk does generate value as the connections obtained are hooked elsewhere. However I still think the random walking doesn't belong in the bootstrap, and should be an explicit FindPeers wrapper elsewhere. Meaning routing table health is maintained with periodic `FindPeer(self)`, and filecoin "mesh" health generated by periodic `FindPeer(random)`.

Reply:
Mind opening an issue in libp2p/specs with your thoughts to continue discussing there? This is likely not the right place for the kind of engagement the conversation requires.